File: codetree/trail2/짝수만 2로 나누기/divide-even-numbers-by-2.ts
declare var require: any;
const fs: any = require("fs");

const filePath: string = fs.existsSync("./input_test.txt")
    ? "./input_test.txt"
    : "/dev/stdin";

const input: string[] = fs.readFileSync(filePath, "utf-8").trim().split(/\n+/);

/* 📥 Input */
const getInputData = (): [number, number[]] => {
    const n = Number(input[0])
    const numbers: number[] = input[1].split(' ').map(Number);
  
    return [n, numbers];
};

/* ⚙️ Logic */
const solution = (n: number, numbers: number[]): number[] => {
    const answer : number[] = [];
    
    for (const number of numbers) {
        if (number % 2 === 0) {
            answer.push(number / 2);
        }
        else {
            answer.push(number);
        }
    }

    return answer;
};

/* 🚀 Run Program */
(() => {
    const [n, numbers] = getInputData();
    console.log(solution(n, numbers).join(' ')); 
})();

File: codetree/trail2/짝수만 2로 나누기/divide-even-numbers-by-2.py
import os
import sys

file_path = os.path.join(os.path.dirname(__file__), "input_test.txt")

if os.path.exists(file_path):
    sys.stdin = open(file_path, "r", encoding="utf-8")


# 📥 Input
def get_input_data() -> tuple[int, list[int]]:
    n = int(input().strip())
    numbers = list(map(int, input().split()))

    return n, numbers


# ⚙️ Logic
def solution(n: int, numbers: list[int]) -> tuple[int, list[int]]:
    answer = []
    for number in numbers:
        if(number % 2 == 0):
            answer.append(number // 2)
        else:
            answer.append(number)

    return answer


# 🚀 Run Program
if __name__ == "__main__":
    n, numbers = get_input_data()
    print(*solution(n, numbers))